storageconfig: encryption YAML support and validation

RaduBerinde opened the pull request:
#### storageconfig: move StoreEncryptionSpec to cli

The separate encryption specs are a byproduct of having to use a
separate command-line flag. This change moves the related code to the
cli package.

Epic: none
Release note: None

#### storageconfig: move WAL failover flag code to cli

Move the pflag-specific part to cli.

Epic: none
Release note: None

#### storageconfig: use time.Duration for rotation period

Epic: none
Release note: None

#### storageconfig: encryption YAML support and validation

Support yaml for the encryption options and add validation.

Epic: none
Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/154579)
<!-- Reviewable:end -->

jbowens reviewed (approved):
<img class="emoji" title=":lgtm_strong:" alt=":lgtm_strong:" align="absmiddle" src="https://reviewable.io/lgtm_strong.png" height="20" width="61"/> 

> @jbowens reviewed 10 of 10 files at r1, 8 of 8 files at r2, 5 of 5 files at r3, 6 of 6 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/154579#-:-OaadHY71SbzmiBhEeOd:b-uwkowd)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @xxmplus)



<!-- Sent from Reviewable.io -->

RaduBerinde commented:
TFTR!

bors r+

RaduBerinde committed:
storageconfig: move StoreEncryptionSpec to cli

The separate encryption specs are a byproduct of having to use a
separate command-line flag. This change moves the related code to the
cli package.

Epic: none
Release note: None

RaduBerinde committed:
storageconfig: use time.Duration for rotation period

Epic: none
Release note: None

RaduBerinde committed:
storageconfig: move WAL failover flag code to cli

Move the pflag-specific part to cli.

Epic: none
Release note: None

RaduBerinde committed:
storageconfig: encryption YAML support and validation

Support yaml for the encryption options and add validation.

Epic: none
Release note: None

RaduBerinde commented:
bors r+
